learning/dashboard/angular_routing/Routing/README.md

```diff
@@ -0,0 +1,122 @@
+
+Methods:
+
+  Parent to Child: via Input,
+
+  Child to Parent: via Output() and EventEmitter,
+
+  Child to Parent: via ViewChild,
+
+  Unrelated Components: via a Service.
+
+
+**Why is data passed between Angular components:**
+
+
+Angular provides component based architecture that allows modularizing the application. It means you can create multiple chunks, and convert your large component to a smaller segment that can be easily maintained. The main advantage to do this is easily understanding the component and maintenance of code. In this scenario, it is important for communication between components or share the data between the component. 
+
+Angular provides multiple ways to share the data between the components. Based on your need and kind of component relation, you can select any of the methods. It is recommended to use the service method as it is independent of component relation and helps you to send data to multiple components. 
+
+
+**How to pass the data between angular  routes?**
+
+
+The task is to share data variables between two or more controllers by using AngularJS. There are many procedures to achieve this. Here we will discuss the most popular ones. 
+
+Approach: 
+
+To share data between the controllers in AngularJS we have two main cases:
+
+Share data between parent and child: Here, the sharing of data can be done simply by using controller inheritance as the scope of a child controller inherits from the scope of the parent controller.
+Share data between controllers without having a relation: Here, the sharing of data can be done in a few ways some of them are: 
+
+By using the rootScope variable:
+
+We can use the rootScope variable to hold shared data and then can reference it from any controller. Here, at the start of the Angular app, we initialized the rootScope variable with some value and then refer it from every controller and thus binding scope variables in both controllers to the rootScope variable.
+
+By using factory or service:
+
+The $rootscope method is not preferred for data transfer or sharing data because it has a global scope that is available for the entire application. So, we use another method in which we create a factory or service to hold share data. AngularJS factories and services are JS functions that perform a specific task containing both methods & properties and can be injected into other components (e.g. your controllers) using dependency injection. In this way we can define a shared variable in a factory, inject it into both controllers and thus bind scope variables in both controllers to this factory data.
+
+
+# TUTORIAL APP
+
+
+Link for a simple Angular application to demonstrate routing: <br>
+
+(https://github.com/ShambhaviVijay/AngularAce-))
+
+
+# REFERENCES:
+
+(https://www.geeksforgeeks.org) <br>
```

Comment:
Use markdown syntax, as explained above, for all external links. Provide a meaningful title and then the URL.